docs/nlu/using-bert.rst

```diff
@@ -0,0 +1,159 @@
+Notice that we add two additional components :ref:`LanguageModelTokenizer` and :ref:`LanguageModelFeaturizer` which
+pick up the tokens and feature vectors respectively that are constructed by the utility component.
+
+We use the same :ref:`diet-classifier` model for combined intent classification and entity recognition in both cases.
+
+Run the Pipelines
+-----------------
+
+You can run both configurations yourself.
+
+.. code-block:: yaml
+
+    mkdir gridresults
+    rasa test nlu --config configs/config-light.yml \
+                  --cross-validation --runs 1 --folds 2 \
+                  --out gridresults/config-light
+    rasa test nlu --config configs/config-heavy.yml \
+                  --cross-validation --runs 1 --folds 2 \
+                  --out gridresults/config-heavy
+
+When this runs you should see logs appear. We've picked a few
+of those lines to list them here. 
+
+.. code-block:: txt
+
+    # output from the light model
+    2020-03-30 16:21:54 INFO     rasa.nlu.model  - Starting to train component DIETClassifier
+    Epochs: 100%|███████████████████████████████| 50/50 [04:30<00:00, ...]
+    2020-03-30 16:23:53 INFO     rasa.nlu.test  - Running model for predictions:
+    100%|███████████████████████████████████████| 2396/2396 [01:23<00:00, 28.65it/s]
+    ...
+    # output from the heavy model
+    2020-03-30 16:47:04 INFO     rasa.nlu.model  - Starting to train component DIETClassifier
+    Epochs: 100%|███████████████████████████████| 50/50 [04:33<00:00,  ...]
+    2020-03-30 16:49:52 INFO     rasa.nlu.test  - Running model for predictions:
+    100%|███████████████████████████████████████| 2396/2396 [07:20<00:00,  5.69it/s]
+
+From the logs we can gather an important observation. 
+The heavy model is a fair bit slower, not in training, but at inference time
+we see a ~6 fold increase. Depending on your use-case this is 
+something to seriously consider.
+
+Results
+-------
+
+We've summarised the results into two charts, one for intents and
+one for entities.
+
+
+Intent Results 
+~~~~~~~~~~~~~~
+
+.. image:: /_static/images/bert-intents.png
+
+Entity Results 
+~~~~~~~~~~~~~~
+
+.. image:: /_static/images/bert-entities.png
+
+Observations 
+~~~~~~~~~~~~
+
+On all fronts we see that the model with the Bert embeddings performs better. 
+But it deserves mentioning that the effect is more pronounced in the entities.
+Note that these results may not be the same on your use-case. Every assistant 
+is different so it is important that you keep comparing. 
+
+It also deserves 
+mentioning that you need to beware that you don't over-optimise training data
+that you've generated yourself. End users will use the assistant in ways you 
+probably did not anticipate. Typically it is more important to gather data of 
+actual users than it is to get the best F1 score on an artificial dataset.
```

Comment:
Is this related to this doc page? I didn't understand what you mean here as well.

Reply:
rephrased it.